### PR TITLE
support verbose loggin of curl request and JSON reply

### PR DESCRIPTION
**Pull request by tpo**
This add a `-verbose` flag, respectively a `verbose: true` setting in the config.yml file.

If set then both the issued curl command with command line options and the JSON reply from the server are logged (password is XXXXed out).

Please pull

By the way, are you not interested in enabling [crypted, authenticated etcd tls connections](https://github.com/cybertec-postgresql/vip-manager/pull/28)? There hasn't been any feedback to that pull request?

### CODE REVIEW

**Comment by tpo** [comment]
I've also added commit ac4419d which adds a few line breaks for (IMHO) improved code readability. Pull that additional commit as you wish -

File: hetznerConfigurer.go [hunk 1]
```diff
@@ -29,10 +29,15 @@ type HetznerConfigurer struct {
 	*IPConfiguration
 	cachedState  int
 	lastAPICheck time.Time
+	verbose      bool
 }
 
-func NewHetznerConfigurer(config *IPConfiguration) (*HetznerConfigurer, error) {
-	c := &HetznerConfigurer{IPConfiguration: config, cachedState: UNKNOWN, lastAPICheck: time.Unix(0, 0)}
+func NewHetznerConfigurer(config *IPConfiguration, verbose_ bool) (*HetznerConfigurer, error) {
```
Comment (pashagolub) [inline hetznerConfigurer.go:35]:
```suggestion
func NewHetznerConfigurer(config *IPConfiguration, verbose bool) (*HetznerConfigurer, error) {
```

File: hetznerConfigurer.go [hunk 2]
```diff
@@ -29,10 +29,15 @@ type HetznerConfigurer struct {
...
+	c := &HetznerConfigurer{
+		IPConfiguration: config,
+		cachedState: UNKNOWN,
+		lastAPICheck: time.Unix(0, 0),
+		verbose: verbose_}
```
Comment (pashagolub) [inline hetznerConfigurer.go:40]:
```suggestion
		verbose: verbose}
```

**Comment by tpo** [comment]
> Looks good to me. Why `verbose_` instead of `verbose`?

No reason I can think of. If it compiles without the underscore then all is good. Thanks @pashagolub !

**Comment by markwort** [comment]
dear @tpo ,
I took your changes and rebased them against the current master, which includes new config handling.
Could you verify that everything still works? ( I have no access to the Hetzner API for such tests at the moment...)

I have create a new branch: tpo-add_verbose_curl_logging
If you've succesfully tested it, I think you could take that branch and rebase the one in your repository with it, then it should show up in this PR as well.
Otherwise, if you don't mind, we could just merge the tpo-add_verbose_curl_logging branch and close this merge request with an appropriate message...

Looking forward to your feedback.

**Comment by tpo** [comment]
> dear @tpo ,
> I took your changes and rebased them against the current master, which includes new config handling.
> Could you verify that everything still works? ( I have no access to the Hetzner API for such tests at the moment...)
> 
> I have create a new branch: tpo-add_verbose_curl_logging
> If you've succesfully tested it, I think you could take that branch and rebase the one in your repository with it, then it should show up in this PR as well.
> Otherwise, if you don't mind, we could just merge the tpo-add_verbose_curl_logging branch and close this merge request with an appropriate message...
> 
> Looking forward to your feedback.

Sorry it took me this long to reply.

I think all is fine here right @markwort ? You do not need any feedback from me any more on this topic, correct?

**Comment by markwort** [comment]
Yes, I already went ahead and brought your changes into master.
I had no change of actually testing the code, but it seemed like you really only added a few line breaks in between arguments and the logging calls themselves, so I don't suppose there is a way that this change was able to break anything.

**Pull request merged**